sdk/core/System.ClientModel/src/Message/BinaryContent.cs

```diff
@@ -105,8 +105,11 @@ private ModelWriter Writer
                     throw new InvalidOperationException("Cannot use Writer with non-IJsonModel model type.");
                 }
 
-                _writer ??= new ModelWriter((IJsonModel<object>)jsonModel, _options);
-                return _writer;
+                if (_sequenceReader == null)
+                {
+                    _sequenceReader = new ModelWriter<T>(jsonModel, _options).GetSequenceBufferReader();
```

Comment:
Can we call the method ExtractReader or something else with "Extract" in the name. I want to make sure callers understand that the ownership of the buffers is transferred and Get does not communicate it.